cpp/ql/src/change-notes/2023-04-11-use-after-free.md

```diff
@@ -0,0 +1,4 @@
+---
+category: newQuery
+---
+* The query `cpp/use-after-free` has been modernized and assigned the precision "medium". The query finds cases of where a pointer is dereferenced after its memory has been deallocated.
```

Comment:
Should the new query be `cpp/double-free`?

Reply:
This PR contains two new queries: A new query with the id `cpp/double-free` (whose change note is added [here](https://github.com/github/codeql/pull/12569/commits/c1960c6ff91141bbc47e6791a0f63af25c40d480)) and another query `cpp/use-after-free` (whose change note is added [here](https://github.com/github/codeql/pull/12569/commits/259d5b64525256ff5d03d9842b6d474f74a1230b)). So I think this is correct.